glean-core/src/dispatcher/mod.rs

```diff
@@ -135,20 +138,16 @@ impl DispatchGuard {
 
     fn send(&self, task: Command) -> Result<(), DispatchError> {
         if self.queue_preinit.load(Ordering::SeqCst) {
-            match self
-                .preinit_sender
-                .try_send(task)
-                .map_err(DispatchError::from)
-            {
-                Err(err @ DispatchError::QueueFull) => {
-                    // This value ends up in the `preinit_tasks_overflow` metric, but we
-                    // can't record directly there, because that would only add
-                    // the recording to an already-overflowing task queue and would be
-                    // silently dropped.
-                    self.overflow_count.fetch_add(1, Ordering::SeqCst);
-                    Err(err)
-                }
-                err => err,
+            if self.preinit_sender.len() < self.max_queue_size {
+                self.preinit_sender
+                    .try_send(task)
+                    .map_err(DispatchError::from)
+            } else {
+                self.overflow_count.fetch_add(1, Ordering::SeqCst);
+                // Instead of using a bounded queue, we are handling the bounds
+                // checking ourselves. A bounded queue would return a QueueFull
+                // DispatchError if full, so we do the same.
```

Comment:
Actually we translate the send error to a `QueueFull` ourselves here: https://github.com/mozilla/glean/blob/main/glean-core/src/dispatcher/mod.rs#L92
Which means we could now remove that `from` implementation and above use `send`, not `try_send` (as we can't hit a full queue there anyway)

Reply:
👍 ok I switched it over and updated the comment to be more accurate, lmk if any further changes are necessary

Reply:
you can now remove these lines: https://github.com/mozilla/glean/blob/main/glean-core/src/dispatcher/mod.rs#L89-L96

Reply:
updated, also removed the now unused import.